### ISSUE TITLE
Graphic bug and freeze

### ISSUE THREAD

**Issue by the-gods-are-cruel** [ValveSoftware/Dota-2#613]
in a bot game for about 15s in desktop friendly fullscreen when this happened: http://i.imgur.com/RGGd0Rq.jpg

**Comment by Tele42** [comment]
Please include your system information as requested at https://github.com/ValveSoftware/Dota-2#system-information.

**Comment by the-gods-are-cruel** [comment]
here you go:

Processor Information:
    Model:  iMac13,1
    Vendor:  GenuineIntel
    CPU Family:  0x6
    CPU Model:  0x3a
    CPU Stepping:  0x9
    CPU Type:  0x0
    Speed:  2700 Mhz
    4 logical processors
    4 physical processors
    HyperThreading:  Unsupported
    FCMOV:  Supported
    SSE2:  Supported
    SSE3:  Supported
    SSSE3:  Supported
    SSE4a:  Unsupported
    SSE41:  Supported
    SSE42:  Supported

Network Information:
    Network Speed:  

Operating System Version:
    MacOS 10.11.1 (64 bit)

Video Card:
    Driver:  NVIDIA GeForce GT 640M

```
Driver Version:  1.2
Driver Date: 0  0
OpenGL Version: 2.1
Desktop Color Depth: 32 bits per pixel
Monitor Refresh Rate: 0 Hz
VendorID:  0x10de
DeviceID:  0xfd8
Number of Monitors:  1
Number of Logical Video Cards:  1
Primary Display Resolution:  1920 x 1080
Desktop Resolution: 1920 x 1080
Primary Display Size: 18.62" x 10.47"  (21.34" diag)
                                        47.3cm x 26.6cm  (54.2cm diag)
Primary VRAM: 512 MB
```

Memory:
    RAM:  8192 Mb

Miscellaneous:
    UI Language:  English
    Microphone:  Not set
    Media Type:  DVD
    Total Hard Disk Space Available:  953049 Mb
    Largest Free Hard Disk Block:  834210 Mb

Installed software:
    Accessibility Inspector
    Agent
    App Store
    Application Loader
    Automator
    Battle.net
    Blizzard Error
    Calendar
    Contacts
    Diablo III
    Diablo III Launcher
    FaceTime
    FileMerge
    Flux
    Game Center
    Gmail
    Google Chrome
    Google Drive
    Google Hangouts
    Google Search
    HearthStats
    Hearthstone
    Hearthstone Beta Launcher
    Heroes
    Heroes of the Storm
    HeroesSwitcher
    iBooks
    Image Capture
    Install OS X El Capitan
    Instruments
    Launchpad
    League of Legends
    Maps
    Messages
    MiniTerm
    Mission Control
    Notes
    Photos
    Reminders
    SC2
    SC2Switcher
    StarCraft II
    StarCraft II Editor
    Steam
    Stickies
    TouchFly
    World of Warcraft
    World of Warcraft Launcher
    Xcode
    YouTube

**Comment by danginsburg** [comment]
Does this happen with the NVIDIA web driver: http://www.nvidia.com/download/driverResults.aspx/93460/en-us .  I have not seen this, it looks like potentially a driver issue.  Does the game crash when this occurs?

**Comment by CalMlynarczyk** [comment]
I have been dealing with these multiple times per game. It is especially bad when watching replays, where they occur every couple of minutes. I have included both the system and user error logs from the most recent crash I had.

[OS X Error Logs](https://gist.github.com/CalMlynarczyk/649ba2a2c1c45477c295)

**Comment by CalMlynarczyk** [comment]
I am having a [similar issue](http://steamcommunity.com/app/255710/discussions/0/517142892069652728/) with _Cities: Skylines_ lately as well.

**Comment by danginsburg** [comment]
@CalMlynarczyk Your crash is inside the NVIDA OpenGL driver.  Please try installing the NVIDIA web driver and report back if it fixes the problem: http://www.nvidia.com/download/driverResults.aspx/93460/en-us

Thread 9 Crashed:
0   com.apple.GeForceGLDriver       0x00007fff9818ac0c 0x7fff97e62000 + 3312652
1   com.apple.GeForceGLDriver       0x00007fff98084957 gldGetQueryInfo + 38
2   GLEngine                        0x00007fff8708878a glGetQueryObject_Core + 254
3   GLEngine                        0x00007fff8708889a glGetQueryObjectuiv_Exec + 37
4   librendersystemgl.dylib         0x00000000884a69c7 CGLMQuery::Complete(void_, int) + 103
5   librendersystemgl.dylib         0x00000000884a70cf CGLMQuery::GetData(void_, unsigned int, bool) + 687
6   librendersystemgl.dylib         0x00000000884875e3 CRenderThreadGL::ManageQueries(bool) + 99
7   librendersystemgl.dylib         0x000000008848abfc CRenderThreadGL::PresentInternal(CommandPresentGL_t const&) + 844
8   librendersystemgl.dylib         0x0000000088489b0e CRenderThreadGL::Execute(CCommandStream_, bool) + 414
9   librendersystemgl.dylib         0x0000000088457379 CRenderThreadBase::Dispatch(CCommandStream_, bool) + 137
10  librendersystemgl.dylib         0x0000000088457a2a CRenderThreadBase::Run() + 346
11  librendersystemgl.dylib         0x0000000088479c0d RunOpenGLThread(void_) + 189
12  libtier0.dylib                  0x00000000108725c9 ThreadProcConvert(void_) + 73
13  libsystem_pthread.dylib         0x00007fff882039b1 _pthread_body + 131
14  libsystem_pthread.dylib         0x00007fff8820392e _pthread_start + 168
15  libsystem_pthread.dylib         0x00007fff88201385 thread_start + 13

**Comment by CalMlynarczyk** [comment]
@danginsburg I installed the web driver, but it is causing more problems than it fixed. In _ALL_ games, I am getting major keyboard and mouse input issues where only about 25% of input commands register. In Dota 2, it is even worse, where the game will go multiple seconds without accepting any user input. At one point, it gave up entirely and would not respond to any user input until I restarted the game. The performance was also seriously degraded; I had to set the resolution down to 1376x800 (or whatever that resolution is) just to get the game to 30 FPS, whereas before I could run it at 1920x1080 at 60+ FPS (without changing any other settings).

After reverting back to the built-in Mac driver, I am no longer seeing these issues anymore (sans the original driver crash issue). I am tracking that this is a driver issue, and I can bring it up directly with Nvidia and Apple since Valve can't really do anything about it. However, you guys might get a better response if you report it to them, as I imagine you have some close points of contact at Nvidia.
